WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift

```diff
@@ -629,6 +651,56 @@ private extension OrderDetailsViewController {
         present(navigationController, animated: true, completion: nil)
     }
 
+    func trashOrderTapped() {
+        ServiceLocator.analytics.track(.orderDetailTrashButtonTapped)
+
+        let alertController = UIAlertController(title: Localization.Alert.orderTrashConfirmationTitle,
+                                                message: Localization.Alert.orderTrashConfirmationMessage,
+                                                preferredStyle: .alert)
+        let cancel = UIAlertAction(title: Localization.Alert.orderTrashConfirmationCancelButton,
+                                   style: .cancel) { (action) in
+        }
+        let confirm = UIAlertAction(title: Localization.Alert.orderTrashConfirmationConfirmButton,
+                                    style: .default) { [weak self] (action) in
+            self?.trashOrderAction()
+        }
+        alertController.addAction(cancel)
+        alertController.addAction(confirm)
+        present(alertController, animated: true)
+    }
+
+    func trashOrderAction() {
+        let order = viewModel.order
+        viewModel.trashOrder { [weak self] result in
+            switch result {
+            case .success:
+                NotificationCenter.default.post(name: .ordersBadgeReloadRequired, object: nil)
+                self?.displayOrderTrashUndoNotice {
+                    self?.undoTrashOrderAction()
+                }
+
+                // Navigate back to the master view controller of the split view controller to display the order list.
+                if let splitViewController = self?.splitViewController,
```

Comment:
For my learning: I wonder how this `splitViewController` works correctly on iPhone. Are we supposed to use `splitViewController` to get the `navigationController` on screens in which we support split screen?